riscv/triggers.cc

```diff
@@ -106,6 +106,21 @@ bool trigger_t::textra_match(processor_t * const proc) const noexcept
   return true;
 }
 
+bool allow_action(state_t * const state, const action_t action)
```

Comment:
Any reason this isn't a method on `trigger_t`? Then you wouldn't have to pass in `action`

Reply:
Move the allow_action() function to trigger_t.